src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/CustomerThreadController.php

```diff
@@ -66,13 +72,42 @@ class CustomerThreadController extends FrameworkBundleAdminController
      */
     public function indexAction(Request $request, CustomerThreadFilter $filters): Response
     {
+        $customerThreadKpiFactory = $this->get(HookableKpiRowFactory::class);
         $customerThreadGridFactory = $this->get('prestashop.core.grid.factory.customer_thread');
         $customerThreadGrid = $customerThreadGridFactory->getGrid($filters);
 
+        $contactOptionType = $this->createForm(ContactOptionType::class, [
+            'PS_CUSTOMER_SERVICE_FILE_UPLOAD' => $this->getConfiguration()->get('PS_CUSTOMER_SERVICE_FILE_UPLOAD'),
+            'PS_CUSTOMER_SERVICE_SIGNATURE' => $this->getConfiguration()->get('PS_CUSTOMER_SERVICE_SIGNATURE'),
+        ]);
+
+        /** @var CustomerThreadRepository $customerThreadRepository */
+        $customerThreadRepository = $this->get(CustomerThreadRepository::class);
+        /** @var CustomerMessageRepository $customerMessageRepository */
+        $customerMessageRepository = $this->get(CustomerMessageRepository::class);
+
+        $statisticRows = [
+            $this->trans('Total threads', 'Admin.Catalog.Feature') => $all = $customerThreadRepository->getTotalCustomerThreads(),
+            $this->trans('Threads pending', 'Admin.Catalog.Feature') => $pending = $customerThreadRepository->getTotalCustomerThreads('status LIKE "%pending%"'),
+            $this->trans('Total number of customer messages', 'Admin.Catalog.Feature') => $customerMessageRepository->getTotalCustomerMessages('id_employee = 0'),
+            $this->trans('Total number of employee messages', 'Admin.Catalog.Feature') => $customerMessageRepository->getTotalCustomerMessages('id_employee != 0'),
+            $this->trans('Unread threads', 'Admin.Catalog.Feature') => $unread = $customerThreadRepository->getTotalCustomerThreads('status = "open"'),
+            $this->trans('Closed threads', 'Admin.Catalog.Feature') => $all - ($unread + $pending),
+        ];
```

Comment:
I don't think this should be done in the controller, instead you should handle this via a query like you did for `GetCustomerServiceSummary`

OR

you could integrate these statistics in the same CQRS query since they're very related, you'll just have to adapt the returned QueryResult (the query name should also be renamed if you choose to adapt the query result):

```
CustomerServiceDetails
- summaries: CustomerServiceSummary[]
- statistics: array<string, int>
```

Reply:
Reminder, this part should be handled in the CQRS query